src/System.Memory/tests/Span/Clear.cs

```diff
@@ -198,32 +198,40 @@ public unsafe static void ClearLongerThanUintMaxValueBytes()
         {
             if (sizeof(IntPtr) == sizeof(long))
             {
-                // The maximum index in any single dimension is 2,147,483,591 (0x7FFFFFC7) 
-                // for byte arrays and arrays of single-byte structures, 
-                // and 2,146,435,071 (0X7FEFFFFF) for other types.
-                const int maxArraySizeForLargerThanByteTypes = 0X7FEFFFFF;
-                var a = new int[maxArraySizeForLargerThanByteTypes];
-
-                int initial = 5;
-                for (int i = 0; i < a.Length; i++)
+                try
                 {
-                    a[i] = initial;
-                }
+                    // The maximum index in any single dimension is 2,147,483,591 (0x7FFFFFC7) 
+                    // for byte arrays and arrays of single-byte structures, 
+                    // and 2,146,435,071 (0X7FEFFFFF) for other types.
+                    const int maxArraySizeForLargerThanByteTypes = 0X7FEFFFFF;
+                    var a = new int[maxArraySizeForLargerThanByteTypes];
 
-                var span = new Span<int>(a);
+                    int initial = 5;
+                    for (int i = 0; i < a.Length; i++)
+                    {
+                        a[i] = initial;
+                    }
 
-                // Act
-                span.Clear();
+                    var span = new Span<int>(a);
 
-                // Assert using custom code for perf and to avoid allocating extra memory
-                for (int i = 0; i < a.Length; i++)
-                {
-                    var actual = a[i];
-                    if (actual != 0)
+                    // Act
+                    span.Clear();
+
+                    // Assert using custom code for perf and to avoid allocating extra memory
+                    for (int i = 0; i < a.Length; i++)
                     {
-                        Assert.Equal(0, actual);
+                        var actual = a[i];
+                        if (actual != 0)
+                        {
+                            Assert.Equal(0, actual);
+                        }
                     }
                 }
+                // Skipping test if Out-of-Memory, since this test can only be run, if there is enough memory
+                catch (OutOfMemoryException)
```

Comment:
Could we tighten up the catch block to just be around the array allocation?  I assume that's the only place for which we expect/suppress this.